localgov_alert_banner.page.inc

```diff
@@ -35,8 +35,11 @@ function template_preprocess_localgov_alert_banner(array &$variables) {
 
   // Set default link text if field exists.
   if ($entity->hasField('link')) {
-    if ($entity->get('link')->title === '') {
-      $variables['content']['link'] = Link::fromTextAndUrl(t('More information'), Url::fromUri($entity->get('link')->uri))->toString();
+    $link = $entity->get('link');
+    if ($link->title === '') {
+      $link_value = $link->getValue();
+      $link_value[0]['title'] = t('More information');
+      $entity->set('link', $link_value);
```

Comment:
I don't think this will work, as when altering entities, even right before display, we still have to do an entity->save() operation. But this would happen everytime it's displayed.

So we would either
- Move this to a preSave hook / event. This means it will add the text as the link text 'More information' and would not be changed if we ever updated the default
- Move this to a preprocess_link function, if such a function exists where the link text can be altered directly without saving the text.

I'm open to either, there are pros and cons to each, and it depends on how common it is for content designers to leave this blank.

I wonder if its possible to have default link text in Drupal core that is then overrideable.

Reply:
> I don't think this will work, as when altering entities, even right before display, we still have to do an entity->save() 

This seems obvious when you say it :facepalm:.

On the other hand, It is definitely working for me right now with an unmodified `localgov_base` theme (authenticated, anonymous, any way I check). I'll have a look tomorrow and see if I can find out the reason (or the source of my hallucination...)

Reply:
> Move this to a preprocess_link function

Preprocess is an option too via `template_preprocess_field()`. With link fields when there's no `#title` provided `#url` and `#title` are identical, so it's possible to compare and override the `#title` value if they're the both set and equivalent.

Reply:
I'd be tempted to go with a `template_preprocess_field()` in that case. Still showing as bare link when I try and on fresh installs.

Reply:
After having a play around with what we have last night I concur with `template_preprocess_field()` route, i'll get the code sorted for this.

Reply:
<img width="837" alt="Welcome____Drush_Site-Install" src="https://github.com/localgovdrupal/localgov_alert_banner/assets/1036167/96ee2bb0-d2ad-4372-a43a-3ccc818ad344">

`/**
 * Implements hook_preprocess_field().
 */
function localgov_alert_banner_preprocess_field(&$variables) {
  if ($variables['element']['#field_name'] == 'link' && $variables['element']['#bundle'] == 'localgov_alert_banner') {
    foreach ($variables['element']['#items'] as $delta => $item) {
      $link_item = $item->getValue();
      if (empty($item->getValue()['title'])) {
        $default_text = t('More information');
        $variables['items'][0]['content']['#title'] = $default_text;
      }
    }
  }
}
`